#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.8884465550270391 0.42776385971423053 0.16637547650711748 5.422701183980594
  position -1.0374639259010932 2.7746778919788926 2.5432561127553837
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorTileSize 0.25 0.25
  wallHeight 0.05
}
WoodenBox {
  translation 0.205843 0.05 -0.221732
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.1926 0.05 -0.265934
  name "wooden box(2)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.12104 0.05 0.28743
  name "wooden box(1)"
  size 0.1 0.1 0.1
}
Robot {
  translation 0 0.04 0
  children [
    Robot {
      translation 0 0 0.1
      children [
        DistanceSensor {
          children [
            DEF SENSOR Shape {
              appearance PBRAppearance {
                baseColor 0.203922 0.396078 0.643137
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
          name "ds_center"
          boundingObject USE SENSOR
        }
      ]
      boundingObject USE SENSOR
      physics Physics {
      }
      controller "DSPrint"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint DEF WHEEL4 Solid {
        translation -0.06 0 -0.05
        rotation 0 0 1 1.5708
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid3"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint DEF WHEEL3 Solid {
        translation 0.06 0 -0.05
        rotation 0 0 1 1.5708
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid2"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint DEF WHEEL2 Solid {
        translation -0.06 0 0.05
        rotation 0 0 1 1.57079
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        name "solid1"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.06 0 0.05
        rotation 0 0 1 1.5708
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "4WheelForward"
}
